engines/sci/graphics/text16.cpp

```diff
@@ -340,6 +357,46 @@ int16 GfxText16::GetLongest(const char *&textPtr, int16 maxWidth, GuiResourceId
 	return resultCharCount;
 }
 
+// GetLongest() originally contained an off-by-one character bug which was
+//  fixed in late 1990. As part of this change, a width test was changed
+//  from greater than or equals to greater than. This method returns which
+//  behavior to use based on the current version and game.
+bool GfxText16::useEarlyGetLongestCalculations() {
+	switch (getSciVersion()) {
+
+	// All SCI0, confirmed:
+	// - LSL2 English PC 1.000.011
+	// - LB1 PC 1.000.046
+	// - ICEMAN PC 1.033
+	// - SQ3 English PC 1.018
+	// - PQ2 Japanese 1.000.052
+	case SCI_VERSION_0_EARLY:
+	case SCI_VERSION_0_LATE:
+		return true;
+
+	// SCI01: confirmed KQ1 and QFG1 Japanese,
+	// fixed in SQ3 and LSL3 multilingual PC
```

Comment:
Perhaps it’s possible to check for changes to core game scripts, instead of hardcoding game IDs? Other checks in features.cpp have been implemented like this